lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -787,7 +787,14 @@ MStatus MayaUsdProxyShapeBase::computeInStageDataCached(MDataBlock& dataBlock)
                 if (nullptr == rootLayer)
                     rootLayer = SdfLayer::FindOrOpen(fileString);
 
-                if (rootLayer) {
+                if (nullptr == rootLayer) {
+                    // Create a new stage in memory with an anonymous root layer.
+                    if (!_anonymousRootLayer)
+                        _anonymousRootLayer = SdfLayer::CreateAnonymous(kAnonymousLayerName);
+                    rootLayer = _anonymousRootLayer;
```

Comment:
Your comment says "create a new stage in memory". Can you point me to the code in USD where that happens. I had a look at SdfLayer::CreateAnonymous() and don't see anywhere that it (or called methods) create a stage in memory.

And I don't fully understand why in the case where we just want an empty layer we would want to fall into the code below and call UsdStage::Open?

Reply:
You are right, the comment should say create a new *layer* in memory!

As for falling down, in the end we want a stage, so for an empty layer, passing the newly-created layer to open will create a stage with that layer. The code is just so that all cases passes in common code paths, to avoid maintenance issues where different ways of creating the stage would not perform the same steps. (For example target the session vs root layer.)

Reply:
Oh, no wait, the comment is correct, but anticipating what will happen: we create an anonymous layer to create an empty stage in memory in the Open call below.